uwblueprint/don-efficace
# Migrate Stripe checkout session from Stripe-hosted to embedded

## Notion ticket link

[Create embedded Stripe checkout session](https://www.notion.so/uwblueprintexecs/create-embedded-Stripe-checkout-session-1b6760ed57a64304ab22bb94008d0afa?pvs=4)



## Implementation description
* Replace previous Stripe-hosted payment page with embedded checkout session



## Steps to test
1. Enter in "42424242..." as the credit card. This is Stripe's "testing" credit card number.
2. Enter in any three-digit CVV, any date in the future, and any 5-digit zip code
3. Enter a dollar amount in the payment field, and click submit
4. Verify that you are redirected to the "checkout-sucess" page
5. Verify that the payment has gone through in Stripe's logs


## What should reviewers focus on?
* We're not sure if the behaviour behind the scenes is correct. It looked like in the original code that a URL was supposed to be returned. However, an ID is instead returned with the way that the backend is setup.
* Our implementation with an embedded checkout session did not require an ID nor a URL, but let us know if that's supposed to be used in some way.


## Checklist
